src/service_args.py

```diff
@@ -93,6 +94,9 @@ def __customize_service_args(self):
         self.__add_firstchain_args(['--node-key-file', utils.NODE_KEY_PATH])
         if self._relay_rpc_urls:
             self.__add_firstchain_args(['--relay-chain-rpc-urls'] + list(self._relay_rpc_urls.values()))
+        if self._chain_spec_url:
+            utils.download_chain_spec(self._chain_spec_url, f'{self.chain_name}.json')
+            self.__replace_chain_name(f'{utils.CHAIN_SPEC_PATH}/{self.chain_name}.json', 0)
```

Comment:
So, since this runs here and also in many of the functions below we could have a case when `__replace_chain_name()` runs twice. Did you test this?

Reply:
Good. Might be some unforeseen consequences.